src/Language/PureScript/TypeChecker/Subsumption.hs

```diff
@@ -60,6 +65,10 @@ subsumes' val (TypeApp f1 r1) (TypeApp f2 r2) | f1 == tyRecord && f2 == tyRecord
     (ts2, r2') = rowToList r2
     ts1' = sortBy (comparing fst) ts1
     ts2' = sortBy (comparing fst) ts2
+  when (r1' == REmpty)
```

Comment:
It took me a little while to see why this works, but I think I understand now. Could you please add a comment along the lines of:

```
-- For { ts1 | r1 } to subsume { ts2 | r2 } when r1 is empty, every property in ts2 must appear in ts1. 
-- If not, then the candidate expression is missing a required property.
-- Conversely, when r2 is empty, even property in ts1 must appear in ts2, or else the expression has
-- an additional property which is not allowed. 
```

Thanks!